### PR TITLE
Make utf8Encode / utf8Decode synchronous. #6139

### LINKED ISSUE #6139: Make utf8Encode  / utf8Decode synchronous.
`utf8Decode/utf8Encode` can be polyfilled with a different  approach when `TextEncoder` is not available to stay synchronous. 

For now we have two new methods in `bytes.js`, `utf8DecodeSync` and `utf8EncodeSync`, we should change all the places the old asynchronous `utf8Decode/utf8Encode` are used to use the sync version and then remove the old ones and rename the sync versions to be just `utf8Decode/utf8Encode`

### PR DESCRIPTION
**Pull request by alabiaga**
Fixes #6139.

- rename synchronous utf8DecodeSync and utf8EncodeSync to utf8Decode and utf8Encode and delete old async methods of that name. Update calls and tests accordingly.

### CODE REVIEW

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 1]
```diff
@@ -799,7 +799,7 @@ export class AmpA4A extends AMP.BaseElement {
           // viewport but cannot wait on promise.  Sadly, need a state a
           // variable.
           this.isVerifiedAmpCreative_ = !!creative;
-          return creative && utf8Decode(creative);
+          return creative && Promise.resolve(utf8Decode(creative));
```
Comment (jridgewell) [inline extensions/amp-a4a/0.1/amp-a4a.js]: This change is unnecessary, the enclosing `then` block will wrap in a promise automatically.

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-fluid.js [hunk 2]
```diff
@@ -148,7 +148,7 @@ describes.realWin('DoubleClick Fast Fetch Fluid', realWinConfig, env => {
 
   it('should style iframe/slot correctly on multi-size creative', () => {
     multiSizeImpl.buildCallback();
-    return utf8Encode('foo').then(creative => {
+    return Promise.resolve(utf8Encode('foo')).then(creative => {
```
Comment (jridgewell) [inline extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-fluid.js]: We can remove this `#then`, making the inner code inline with the above line.

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-fluid.js [hunk 3]
```diff
@@ -165,7 +165,7 @@ describes.realWin('DoubleClick Fast Fetch Fluid', realWinConfig, env => {
 
   it('should have an iframe child with initial size 0x0', () => {
     impl.buildCallback();
-    return utf8Encode('foo').then(creative => {
+    return Promise.resolve(utf8Encode('foo')).then(creative => {
```
Comment (jridgewell) [inline extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-fluid.js]: We can remove this `#then`, making the inner code inline with the above line.

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-fluid.js [hunk 4]
```diff
@@ -202,7 +202,7 @@ describes.realWin('DoubleClick Fast Fetch Fluid', realWinConfig, env => {
         sandbox.spy(impl, 'connectFluidMessagingChannel');
     const onFluidResizeSpy = sandbox.spy(impl, 'onFluidResize_');
     impl.attemptChangeHeight = () => Promise.resolve();
-    return utf8Encode(rawCreative).then(creative => {
+    return Promise.resolve(utf8Encode(rawCreative)).then(creative => {
```
Comment (jridgewell) [inline extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-fluid.js]: We can remove this `#then`, making the inner code inline with the above line.

File: src/runtime.js [hunk 5]
```diff
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+/** Aaron was here **/
```
Comment (jridgewell) [inline src/runtime.js]: Remove.

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-fluid.js (hunk 4 above)
Comment (alabiaga) [reply extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-fluid.js]: done

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 6]
```diff
@@ -1197,7 +1197,7 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
                   /** @type {?../../../src/service/xhr-impl.FetchResponse} */
                   ({
                     headers,
-                    arrayBuffer: () => utf8Encode(creative),
+                    arrayBuffer: () => Promise.resolve(utf8Encode(creative)),
```
Comment (glevitzky) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: Probably outside the scope of this PR, but we should consider changing the definition of FetchResponse, so we don't have to deal with promises unnecessarily.

**Comment by choumx** [comment]
A couple lint errors left and I'll merge.

**Comment by alabiaga** [comment]
had to change the author for all my commits as the tied author was tied to my initial email address that wasn't CLA approved. 

Ran

https://help.github.com/articles/changing-author-info/

Not sure if this is something we want. Maybe it's safer to throw this away and redo this pull request change.

**Comment by choumx** [comment]
Oops, looks we've pulled in commits from other authors too. I've done this a couple times with sloppy rebasing. Might be easier to just recreate the PR at this point.

**Pull request closed without merging**